src/scripts/move.js

```diff
@@ -0,0 +1,257 @@
+'use strict';
+
+// import
+const {
+  getNotEmptyFields,
+  getNotEmptyFieldsCoords,
+  checkBetween,
+  randomField,
+  checkForMerge,
+  checkForMove,
+  setNewHighestScore,
+} = require('./utils');
+
+const {
+  totalCountOfCells,
+  gameScore,
+  allCells,
+  highestScore,
+  values,
+  messageDuringTheGame,
+  messageAfterLose,
+  winCondition,
+} = require('./variables');
+
+function mainListenerRemover() {
+  document.body.removeEventListener('keyup', arrowsPlayHandler);
+}
+
+function createItemInRandomEmptyField() {
+  const emptyFields = [];
+
+  for (const cell of allCells) {
+    if (cell.classList.length === 1) {
+      emptyFields.push(cell);
+    }
+  }
+
+  // create new item:
+  if (emptyFields.length > 0) {
+    const randomIndex = randomField(0, emptyFields.length - 1);
+    const randomValue = randomField(0, values.length - 1);
+
+    emptyFields[randomIndex]
+      .classList.add('field-cell--' + values[randomValue]);
+
+    emptyFields[randomIndex]
+      .innerText = values[randomValue];
+
+  // if its impossible to create a new item:
+  } else {
+    messageAfterLose.classList.remove('hidden');
+    messageDuringTheGame.classList.add('hidden');
+
+    setNewHighestScore(gameScore, highestScore);
+
+    // mainListenerRemover();
+    document.body.removeEventListener('keyup', arrowsPlayHandler);
+  }
+}
+
+// Main movement algorithm:
+function arrowsPlayHandler(e) {
+  let isMoved = false;
+  const notEmptyFields = getNotEmptyFields(allCells);
+  const notEmptyFieldsPositions = getNotEmptyFieldsCoords(allCells);
+
+  switch (e.key) {
+    case 'ArrowDown':
+      for (const field of notEmptyFields.reverse()) {
+        const fieldPos = field.id.split('-').slice(1);
+
+        for (const cell of [...allCells].reverse()) {
```

Comment:
Well, it's exactly what I meant in previous comment =)
It's a bad practice to mutate incoming objects inside functions, 'cause it can be used somewhere  outside function
In case if there are no other options to do it, then ok, but better to avoid it